File: snippets/js/const/const.js
const number = 10;
const decimal = 2.3;
const string = "Hello there.";              // ≡
const array = [2.3, 'Hello there.', [1, 2], {a: 1, b: 2}];
const dictionary = {number: 2.3, "list": [1, 2], 'values': {a: 1, b: 2}};
console.log(string);  // Hello there.          ≡

try
{
    string = "Goodbye";                     // ≡
}
catch (error)
{
    console.log('can\'t change constant');  // ≡
}

File: snippets/cpp/const/const.cpp
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int number() {
    return 10;
}

double decimal() {
    return 2.3;
}

string _string() {
    return "Hello there.";
}

int *array(int arr[]) {
    return arr;
}

string dictionary() {
    return "Hello there.";
}

int main() {
    cout << _string();  // Hello there. 
    return 0;
}